packages/sdk/src/messages/NomadMessage.ts

```diff
@@ -53,19 +46,28 @@ export class NomadMessage<T extends NomadContext> {
   readonly replica: core.Replica;
 
   readonly context: T;
-  protected eventCache: Partial<ErinMessageResult>;
   protected _confirmAt?: Date; 
 
-  constructor(context: T, dispatch: Dispatch, eventCache?: Partial<ErinMessageResult>) {
+  readonly _backend?: MessageBackend;
+
+  constructor(context: T, dispatch: Dispatch, _backend?: MessageBackend) {
     this.context = context;
+    this._backend = _backend;
     this.message = parseMessage(dispatch.args.message);
     this.dispatch = dispatch;
     this.home = context.mustGetCore(this.message.from).home;
     this.replica = context.mustGetReplicaFor(
       this.message.from,
       this.message.destination,
     );
-    this.eventCache = eventCache || {};
+  }
+
+  get backend(): MessageBackend {
+    return this._backend || this.context._backend;
+  }
+
+  get messageHash(): string {
+    return this.dispatch.args.messageHash
```

Comment:
It is a good question in general, but backend will also become a container. I have a question in return, what is the main Identificator of a NomadMessage: dispatchTx or MessageHash? I see that we use dispatchTx a lot because it is useful to communicate with users, but is that something we want to call an ID?

Reply:
I would say dispatchTx because it's always available and easily accessible. But idk, maybe there is a case for messageHash

Reply:
just thought, within the same transaction there might be multiple Messages dispatched, is that correct?